app/src/main/java/com/io/movio/ui/moviedetail/MovieDetailViewModel.kt

```diff
@@ -3,15 +3,25 @@ package com.io.movio.ui.moviedetail
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.io.movio.data.Result
 import com.io.movio.data.models.Movie
-import com.io.movio.data.repositories.MovieRepository
 import com.io.movio.domain.GetMovieByIdUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 class MovieDetailViewModel : ViewModel() {
-    private val _movie = MutableLiveData<Movie>()
-    val movie: LiveData<Movie> = _movie
+    private val _movie = MutableLiveData<Result<Movie>>()
+    val movie: LiveData<Result<Movie>> = _movie
 
     fun getMovie(id: Int){
-   _movie.value = GetMovieByIdUseCase(MovieRepository)(id)!!
-    }
-}
\ No newline at end of file
+        viewModelScope.launch(Dispatchers.IO) {
+               val result = GetMovieByIdUseCase().execute(id)
+                Result.Success(result)
```

Comment:
The use case is already returning a result object, why do you wrap it in a result again?